mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/DefaultEntityFactory.java

```diff
@@ -347,4 +352,80 @@ private CqlNameGenerator buildCqlNameGenerator(TypeElement classElement) {
     }
     return new TypeMirror[0];
   }
+
+  private boolean isTransient(
+      String propertyName,
+      Set<String> transientProperties,
+      TypeElement classElement,
+      ExecutableElement getMethod,
+      VariableElement field,
+      int partitionKeyIndex,
+      int clusteringColumnIndex) {
+
+    // check if property is present in @TransientProperties
+    Class<?> otherAnnotationClass =
+        partitionKeyIndex >= 0
+            ? PartitionKey.class
+            : clusteringColumnIndex >= 0 ? ClusteringColumn.class : null;
+    if (transientProperties.contains(propertyName)) {
+      if (otherAnnotationClass != null) {
+        context
+            .getMessager()
+            .error(
+                classElement,
+                "Properties included in @%s can't be annotated with @%s.",
+                TransientProperties.class.getSimpleName(),
+                otherAnnotationClass.getSimpleName());
+      }
+      return true;
+    }
+
+    // check if property is annotated with @Transient
+    Transient getterAnnotation = getMethod.getAnnotation(Transient.class);
+    boolean isTransient;
+    Element transientElement = getMethod;
+    isTransient = getterAnnotation != null;
```

Comment:
This is really hard to follow. How about treating the cases separately:
```java
if (getMethod.getAnnotation(Transient.class) != null) {
  checkTransientNotAnnotated(field, getMethod, getMethod);
  return true;
} else if (field != null && (field.getAnnotation(Transient.class) != null || field.getModifiers().contains(Modifier.TRANSIENT))) {
  checkTransientNotAnnotated(field, getMethod, field);
  return true;
}
return false;
```

And `checkTransientNotAnnotated` would have a more generic message "Transient properties can't be annotated with %s" to handle both field annotation and modifier.

Reply:
Maybe we can go one step further and extract `field != null && (field.getAnnotation(Transient.class) != null || field.getModifiers().contains(Modifier.TRANSIENT)` condition to separete method to name that check (to make it more readable)